### PR TITLE
added 3 more tests

### PR DESCRIPTION
**Pull request by mrdudz**
this adds 3 more tests to the regression testsuite, including one for the unfixed "equality problem"

### CODE REVIEW

File: test/ref/Makefile [hunk 1]
```diff
@@ -49,8 +49,8 @@ $(WORKDIR)/switch.%rg: CC65FLAGS += -Wc --all-cdecl
 # Also, yacc.c does some things that fail when stack operations are optimized.
 # Therefore, don't optimize them.
 #
```
Comment (greg-king5) [inline test/ref/Makefile]: The comment, too, should disappear.

**Comment by mrdudz** [comment]
indeed :)

**Comment by oliverschmidt** [comment]
Thanks :-)

**Comment by MonteCarlos** [comment]
Are these functions

void __fastcall__ _heapadd (void* mem, size_t size);
/* Add a block to the heap */

size_t __fastcall__ _heapblocksize (const void* block);
/* Return the size of an allocated block */

size_t _heapmemavail (void);
/* Return the total free heap space */

size_t _heapmaxavail (void);
/* Return the size of the largest free block on the heap */

CC65 inventions or do they have another origin like a specific known
c library? If i try a google search using f.e. _heapmaxavail, i get only
CC65 related links which supports the first theory in my opinion.


Yours Monte Carlos

**Comment by polluks** [comment]
In Turbo Pascal "MemAvail" and "MaxAvail" are known
http://putka.upm.si/langref/turboPascal/049E.html
http://putka.upm.si/langref/turboPascal/049B.html

**Comment by MonteCarlos** [comment]
The idea of those functions is quite on the hand. Maybe Uz borrowed
the part of the functions names from Pascal?
However, using these functions should break any compatibility with other
c compilers.

Am 02.07.2015 um 13:33 schrieb polluks:

**Pull request merged**